### ISSUE TITLE
newlineAndIndentContinueMarkdownList isn't compatible with multiplex mode

### ISSUE THREAD

**Issue by thomasjm** [codemirror/CodeMirror#5757]
I'm trying to use `mode/multiplex.js` with Markdown as the outer mode and various other languages as the inner modes. I found that the `newlineAndIndentContinueMarkdownList` function is not multiplex mode aware; see here for how it tries to pull the current state for a given line:

https://github.com/codemirror/CodeMirror/blob/2dab4d527c7a55aabc0a52b28b47ab35b1412ca3/addon/edit/continuelist.js#L23

It assumes that the state is a Markdown mode state on the next line.

I think it would be better if it could detect the presence of multiplex mode and pull out the active mode, by adding something like this right after line 23:

```
// If we're in multiplexing mode, pull out the currently active state.
if (eolState.outer) {
    eolState = eolState.inner || eolState.outer;
}
```

Happy to make a PR if the maintainers agree but wanted to see what you think first.

(Is it possible to make this less fragile by doing a check on the state to see which mode it came from?)

**Comment by marijnh** [comment]
Yes, it would make sense to fix that. You'll want to use `CodeMirror.innerMode` to get at the current mode.

**Comment by thomasjm** [comment]
Fixed by #5761
